### PR TITLE
Attempt to trigger codacy

### PR DESCRIPTION
**Pull request by fregante**
Test for #637 as requested by Codacy’s support

### CODE REVIEW

File: src/blocks/readers/ArrayCompositeReader.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ class ArrayCompositeReader extends Reader {
 
   async isAvailable(): Promise<boolean> {
     const availability = await Promise.all(
-      this._readers.map((x) => x.isAvailable())
+      this._readers.map((y) => y.isAvailable())
```
Comment (fregante) [inline src/blocks/readers/ArrayCompositeReader.ts:57]: `@typescript-eslint/promise-function-async` should have been reported

It used to work: https://github.com/pixiebrix/pixiebrix-extension/pull/553

File: src/contentScript/errors.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@
 import { reportError } from "@/telemetry/logging";
 
 function addErrorListeners(): void {
-  window.addEventListener("error", function (error) {
+  window.addEventListener("error", function (error: any) {
```
Comment (twschiller) [inline src/contentScript/errors.ts:23]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7747711)
```suggestion
  window.addEventListener("error", function (error: unknown) {
```

Comment (twschiller) [inline src/contentScript/errors.ts:23]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7747711)
```suggestion
  window.addEventListener("error", function (error: unknown) {
```

File: src/selectors.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@
     return config;
   }, [options, extensionId, extensionPointId]);
 
-  const [extensionPoint, isPending] = useAsyncState(async () => {
+  const [extensionPoint, isPending] = useAsyncState(() => {
```
Comment (twschiller) [inline src/selectors.ts:66]: Codacy has a fix for the issue: [Functions that return promises must be async.](https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7747711)
```suggestion
  const [extensionPoint, isPending] = useAsyncState(async () => {
```

**Comment by fregante** [comment]
So it seems that Codacy is working:

- it adds annotations on preexisting issues, like https://github.com/pixiebrix/pixiebrix-extension/pull/759/files#diff-c19c7f9cac45e538eee9b1895d91f4447a8a6a9a44f3df87f474058a722d053dR35
- it comments on issues caused by the current PR

All issues of this PR are visible on Codacy: https://app.codacy.com/gh/pixiebrix/pixiebrix-extension/pullRequest?prid=7747711

Screen Shot

**Comment by twschiller** [comment]
Hmm... so the reason why it's passing the PR is because I turned off failing the PR on issues, because IIRC it would also fail sometimes on existing warnings (which was annoying)

image

I was hoping we'd still get a summary comment from Codacy on PRs though:

image

So I guess all-in all, we probably want to turn back on the quality threshold on PRs as our warning stabilize a bit. The alternative is to have to manually view the new warnings reported by Codacy on the Codacy site

Thoughts?

**Comment by fregante** [comment]
Indeed I think that was because it sometimes miscategorizes the issue as “New”

Whether you want to reenable that part is up to you. I think the only difference will be that even if you pre-approve a PR, I will *have to* solve whatever issues pop up before merging it, even if you consider the instances non-urgent or not new, whereas you can always “merge as admin.”

**Comment by twschiller** [comment]
> I will _have to_ solve whatever issues pop up before merging it, even if you pre-approved the PR with non-new or non-urgent Codacy errors

I'll go ahead and re-enable for now and we can re-evaluate the situation. With our current GitHub settings, I think you're right you wouldn't be able to merge an approved pull request with a failing check. I can still merge them though as admin if we disagree with Codacy

**Pull request closed without merging**